### PR TITLE
feat(vercel-kv): support `ttl` for `setItem`

### LINKED ISSUE #266: key expiration support for vercel-kv driver
### Describe the feature

From what I found, current `vercel-kv` driver for unstorage does not provide `ttl` option. Meanwhile the `@vercel/kv` library does provide this option.

https://vercel.com/docs/storage/vercel-kv/kv-reference#set-example

I'm trying to follow the interface used by existing redis driver and implement it in vercel-kv driver. If there are no one working on this, I'm willing to help

my WIP: https://github.com/azophy/unstorage

### Additional information

- [X] Would you be willing to help implement this feature?

### PR DESCRIPTION
**Pull request by azophy**
here we add new option 'ttl' both for driver setup config & for `set()` method. the interface & implementation took notes from existing redis driver.

### 🔗 Linked issue

this resolve #266

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

I have implemented the proposal in #266 . To test it, I have run `pnpm vitest run --coverage` by giving my Vercel KV's credential, and the result shows that it has passed completely. 

image

However because it only uses existing Vercel KV's test suite, it shouldn't have tested the new key expiration functionality. And I'm not sure how I should create the test for this particular feature. I've looked into Redis driver test suite, but I couldn't find section for testing key expiration which I could follow. Maybe other maintainers could give me some pointer? @pi0 @Hebilicious 

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.
- [ ] create appropriate tests

### CODE REVIEW

**Comment by Hebilicious** [comment]
Linking https://github.com/unjs/unstorage/issues/262 for reference. Do we go with this syntax for every driver @pi0 ?

File: src/drivers/vercel-kv.ts [hunk 1]
```diff
@@ -5,8 +5,20 @@ import type { RedisConfigNodejs } from "@upstash/redis";
 import { defineDriver, normalizeKey, joinKeys, createError } from "./utils";
 
 export interface VercelKVOptions extends Partial<RedisConfigNodejs> {
+  /**
+   * Optional prefix to use for all keys. Can be used for namespacing.
+   */
   base?: string;
```
Comment (Hebilicious) [inline src/drivers/vercel-kv.ts:11]: this option is never used

File: src/drivers/vercel-kv.ts [hunk 2]
```diff
@@ -5,8 +5,20 @@ import type { RedisConfigNodejs } from "@upstash/redis";
...
+
+  /**
+   * Optional flag to customzize environment variable prefix (Default is `KV`). Set to `false` to disable env inference for `url` and `token` options
+   */
   env?: false | string;
```
Comment (Hebilicious) [inline src/drivers/vercel-kv.ts:16]: this option is never used too

**Review by Hebilicious — COMMENTED** [review]
I like the default ttl options !

**Comment by azophy** [comment]
Right now, I've just replaced unstorage dependency for one of my NuxtJs project with a github branch based on this PR. I know that this kind of manual testing is not ideal, but currently this is the best approach I could came out with to verify that this code have worked.

my project: https://papan-skor.vercel.app
github repo: https://github.com/azophy/papan-skor
commit that introduce dependency replacement: https://github.com/azophy/papan-skor/commit/67cc078c5318f365e22f5ecc7bb114f5c7452ef5

I verify that the new `ttl` option is indeed worked by manually checking the results of Redis TTL command from my Vercel KV dashboard 

Screenshot from my Vercel KV Dashboard

**Review by pi0 — APPROVED** [review]
Thanks!

**Pull request merged**